src/azure/cli/commands/arm.py

```diff
@@ -150,3 +151,201 @@ def required_values_validator(namespace):
     APPLICATION.remove(APPLICATION.COMMAND_TABLE_LOADED, add_id_parameters)
 
 APPLICATION.register(APPLICATION.COMMAND_TABLE_LOADED, add_id_parameters)
+
+def register_generic_update(name, getter, setter, setter_arg_name='parameters'):
+    from msrestazure.azure_operation import AzureOperationPoller
+    from azure.cli.commands import CliCommand, command_table
+    from azure.cli.commands._introspection import extract_args_from_signature
+
+    get_arguments = dict(extract_args_from_signature(getter))
+    set_arguments = dict(extract_args_from_signature(setter))
+
+    def handler(args):
+        ordered_arguments = args.pop('ordered_arguments')
+
+        getterargs = {key: val for key, val in args.items()
+                      if key in get_arguments}
+        instance = getter(**getterargs)
+
+        # Update properties
+        for arg in ordered_arguments:
+            arg_type, expressions = arg
+            if arg_type == '--set':
+                try:
+                    for expression in expressions:
+                        set_properties(instance, expression)
+                except ValueError:
+                    raise CLIError('--set should be of the form:'
+                                   ' --set property.property=<value>'
+                                   ' property2.property=<value>')
```

Comment:
I agree, but we have to execute the code in order to know if the syntax was correct.  We could do lightweight validation ahead of time, but we would still need these checks for more complicated scenarios

---
In reply to: [72831798](https://github.com/Azure/azure-cli/pull/558#discussion_r72831798) [](ancestors = 72831798)